cassandra-4/src/main/java/io/debezium/connector/cassandra/Cassandra4CommitLogReadHandlerImpl.java

```diff
@@ -288,8 +295,17 @@ private void process(PartitionUpdate pu, OffsetPosition offsetPosition, Keyspace
                         LOGGER.warn("Encountered an unsupported row type {}, skipping...", rowType);
                         continue;
                     }
-                    Row row = (Row) rowOrRangeTombstone;
```

Comment:
Previous code had it wrong. For Cassandra 4, we can not cast `RangeTombstoneBoundMarker` to `Row`. In Cassandra 4, `Row` implements `Unfiltered` and `RangeTombstoneBoundMarker` implements `Unfiltered`, but `RangeTombstoneBoundMarker` does not extend nor implement `Row` so we do not have any information about values but the primary key in `PartitionUpdate`.